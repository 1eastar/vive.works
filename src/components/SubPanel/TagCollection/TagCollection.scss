.items-wrapper {
  display: flex;
  flex-direction: column;
}