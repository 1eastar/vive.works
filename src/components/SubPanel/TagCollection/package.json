{
  "name": "TagCollection",
  "version": "0.0.0",
  "private": true,
  "main": "TagCollection.tsx"
}