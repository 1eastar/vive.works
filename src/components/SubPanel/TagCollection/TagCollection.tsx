/* External */
import { useMemo } from 'react'
import { graphql, useStaticQuery } from 'gatsby'

/* Internal */
import { TagQueryResult } from '@commons/types/QueryType'
import { changeAllMdxToTagCollection } from '@utils/tag.utils'
import TagItem from './TagItem'
import * as styles from './TagCollection.scss'

function TagCollection() {
  const data = useStaticQuery<TagQueryResult>(
    graphql`
      query {
        allMdx {
          nodes {
            frontmatter {
              tags
            }
          }
        }
      }
    `
  )

  const tags = useMemo(() => changeAllMdxToTagCollection(data), [data])

  return (
    <div className={styles.itemsWrapper}>
      { tags.map(TagItem) }
    </div>
  )
}

export default TagCollection
