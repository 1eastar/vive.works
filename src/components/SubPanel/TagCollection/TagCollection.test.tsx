/* External */
import { render, cleanup } from '@testing-library/react'

/* Internal */
import TagCollection from './TagCollection'

afterEach(cleanup)

describe('<TagCollection/>', () => {
  test('성공적으로 렌더링된다.', () => {
    const { TagCollectionEl } = renderTagCollection()

    expect(TagCollectionEl()).toBeInTheDocument()
  })
  
  test('[snapshot] 정확한 Tag 데이터 리스트를 렌더링한다.', () => {
    const { TagCollectionEl } = renderTagCollection()

    expect(TagCollectionEl()).toMatchSnapshot()
  })
})


/* Helper function */
function renderTagCollection() {
  const result = render(<TagCollection/>)

  const TagCollectionEl = () => result.getByTestId('tagCollection')

  return {
    result,
    TagCollectionEl,
  }
}
