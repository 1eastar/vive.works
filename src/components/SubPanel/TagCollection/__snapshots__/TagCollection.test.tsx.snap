// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagCollection/> [snapshot] 정확한 Tag 데이터 리스트를 렌더링한다. 1`] = `
<div
  data-testid="tagCollection"
>
  <div
    class=""
    data-testid="tag-All"
  >
    # 
    All
       (
    5
    )
  </div>
  <div
    class=""
    data-testid="tag-tag1"
  >
    # 
    tag1
       (
    1
    )
  </div>
  <div
    class=""
    data-testid="tag-tag2"
  >
    # 
    tag2
       (
    3
    )
  </div>
  <div
    class=""
    data-testid="tag-tag3"
  >
    # 
    tag3
       (
    2
    )
  </div>
  <div
    class=""
    data-testid="tag-tag4"
  >
    # 
    tag4
       (
    1
    )
  </div>
  <div
    class=""
    data-testid="tag-tag5"
  >
    # 
    tag5
       (
    1
    )
  </div>
</div>
`;
